packages/SidePanel/src/components/Dialog/Dialog.js

```diff
@@ -0,0 +1,78 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { dialogStyles, dialogContentStyles } from "./Dialog.styles";
+
+const propTypes = {
+  children: PropTypes.node.isRequired,
+  footer: PropTypes.node,
+  header: PropTypes.node,
+  isInline: PropTypes.bool,
+  offsetY: PropTypes.number,
+  onAnimationEnd: PropTypes.func.isRequired,
+  onClose: PropTypes.func,
+  refHeader: PropTypes.shape({ current: PropTypes.instanceOf(Element) }).isRequired,
+  refSidePanelContent: PropTypes.shape({ current: PropTypes.instanceOf(Element) }).isRequired,
+  width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
+  isOpen: PropTypes.bool.isRequired,
+};
+
+const defaultProps = {
+  footer: null,
+  header: null,
+  isInline: false,
+  offsetY: 0,
+  onClose: () => {},
+};
+
+function Dialog(props) {
+  const refSidePanel = React.useRef(null);
+
+  const {
+    children,
+    footer,
+    onAnimationEnd,
+    header,
+    isInline,
+    offsetY,
+    onClose,
+    refHeader,
+    refSidePanelContent,
+    width,
+    isOpen,
+    ...moreProps
+  } = props;
+
+  return (
+    <div
+      aria-modal="true"
```

Comment:
Should this be dependent on `isInline` too? If the user can interact with the content below, it's not really a _modal_.